ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/OciConnectorExecutor.java

```diff
@@ -0,0 +1,189 @@
+
+            final String modelResponse = getInputStreamContent((InputStream) response.getEntity());
+            final int statusCode = response.getStatus();
+            if (statusCode < 200 || statusCode >= 300) {
+                throw new OpenSearchStatusException(REMOTE_SERVICE_ERROR + modelResponse, RestStatus.fromCode(statusCode));
+            }
+
+            final ModelTensors tensors = processOutput(modelResponse, connector, scriptService, parameters);
+            tensors.setStatusCode(statusCode);
+            tensorOutputs.add(tensors);
+        } catch (IOException e) {
+            throw new MLException("Fail to execute predict in oci connector", e);
+        }
+    }
+
+    private Response makeHttpCall(String endpoint, String httpMethod, String payload) {
+        final WebTarget target = getWebTarget(endpoint);
+        final WrappedInvocationBuilder wrappedIb = new WrappedInvocationBuilder(target.request(), target.getUri());
+        final Response response;
+        switch (httpMethod.toUpperCase(Locale.ROOT)) {
+            case "POST":
+                response = restClient.post(wrappedIb, payload, new BmcRequest<>());
+                break;
+            case "GET":
+                response = restClient.get(wrappedIb, new BmcRequest<>());
+                break;
+            default:
+                throw new IllegalArgumentException("unsupported http method");
+        }
+        return response;
+    }
+
+    /**
+     *
+     * RestClient is a general wrapper of {@link Client} to talk to OCI services however it has a poor design
+     * and to support calling multiple endpoints. Basically when RestClient::setEndpoint is call it create a
+     * new instance of WebTarget wit that endpoint, but instead of return that instance, customer need to make
+     * another call that WebTarget instance.
+     *
+     * <p>For now let make it synchronize for calling setEndpoint and getBaseTarget to avoid race condition.
+     * Furthermore, it is also pretty cheap compare to calling remote endpoint
+     *
+     * @param endpoint the web target endpoint
+     * @return web target for particular endpoint
+     */
+    @Synchronized
+    private WebTarget getWebTarget(String endpoint) {
+        restClient.setEndpoint(endpoint);
+        return restClient.getBaseTarget();
+    }
+
+    private static BasicAuthenticationDetailsProvider buildAuthenticationDetailsProvider(Connector connector) {
+        final Map<String, String> parameters = connector.getParameters();
+        final OciClientAuthType authType = OciClientAuthType.from(parameters.get(OciConnector.AUTH_TYPE_FIELD).toUpperCase(Locale.ROOT));
+
+        switch (authType) {
+            case RESOURCE_PRINCIPAL:
+                return ResourcePrincipalAuthenticationDetailsProvider.builder().build();
+            case INSTANCE_PRINCIPAL:
+                return InstancePrincipalsAuthenticationDetailsProvider.builder().build();
```

Comment:
add test?

Reply:
Ah this is hard to test in unit test since it will depend on the environment (for example instance principal need to populate in the machine)